notifier/processor.go

```diff
@@ -124,17 +126,36 @@ func (p *Processor) create(ctx context.Context, e Event, prev uuid.UUID) error {
 		return fmt.Errorf("failed to get update diff: %v", err)
 	}
 	log.Debug().Int("removed", len(diff.Removed)).Int("added", len(diff.Added)).Msg("diff results")
-	added, err := p.indexer.AffectedManifests(ctx, diff.Added)
-	if err != nil {
-		return fmt.Errorf("failed to get added affected manifests: %v", err)
+
+	tab := notifTab{
+		N:      make([]Notification, 0),
+		lookup: make(map[string]int),
 	}
-	removed, err := p.indexer.AffectedManifests(ctx, diff.Removed)
-	if err != nil {
-		return fmt.Errorf("failed to get removed affected manifests: %v", err)
+	eg, wctx := errgroup.WithContext(ctx)
+	eg.Go(getAffected(wctx, p.indexer, !p.NoSummary, diff.Added, Added, &tab))
+	eg.Go(getAffected(wctx, p.indexer, !p.NoSummary, diff.Removed, Removed, &tab))
+	if err := eg.Wait(); err != nil {
+		return fmt.Errorf("failed to get affected manifests: %v", err)
 	}
-	log.Debug().Int("added", len(added.VulnerableManifests)).Int("removed", len(removed.VulnerableManifests)).Msg("affected manifest counts")
 
-	if len(added.VulnerableManifests) == 0 && len(removed.VulnerableManifests) == 0 {
+	// Don't count up the affected manifests unless we're going to print it.
+	if ev := log.Debug(); ev.Enabled() {
+		var added, removed int
+		for _, n := range tab.N {
+			switch n.Reason {
+			case Added:
+				added++
+			case Removed:
+				removed++
+			}
+		}
+		ev.
+			Int("added", added).
+			Int("removed", removed).
+			Msg("affected manifest counts")
+	}
+
+	if len(tab.N) == 0 {
```

Comment:
why is this condition added? You're still processing all potential notifications in this method call right?

Reply:
It's the equivalent to `if len(added.VulnerableManifests) == 0 && len(removed.VulnerableManifests) == 0 {`